### DISCUSSION TITLE
Notebook 04: torchmetrics.Accuracy error Section 4: Video 97.

### DISCUSSION THREAD

**Discussion by chriswilson2020** [mrdbourke/pytorch-deep-learning#200] [Q&A] [answered]
In the above video you import torchmetrics 0.7.2 and your code works perfect (once Accuracy is moved to the device)

Your code:
`from torchmetrics import Accuracy`
`torchmetric_accuracy = Accuracy().to(device)`
`torchmetric_accuracy(y_preds, y_blob_test)`

You get the output:
`tensor(0.9950, device='cuda:0')`

However to reproduce your results using torchmetrics 0.11.0 which is the current version in pip I needed to explicitly pass task and num_classes into Accuracy() -> `torchmetric_accuracy = Accuracy(task='multiclass', num_classes=5).to(device)` to obtain the same result. 

If I don't then I get the following error:

---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
[<ipython-input-104-4dd6e691519f>](https://localhost:8080/#) in <module>
      2 
      3 # Setup metric
----> 4 torchmetric_accuracy = Accuracy().to(device)
      5 
      6 # Calculate accuracy

TypeError: __new__() missing 1 required positional argument: 'task'

**I don't know if this is because of something I am doing or there was a change to torchmetrics.Accuracy()**

**Comment by AlessandroMiola** [comment, marked as answer]
There was indeed a major change through these releases which can be tracked at https://github.com/Lightning-AI/metrics/issues/655 and the referenced issues. They added the `task` arg to discern which object to instantiate depending on the task itself (see the issue for the underlying reasons). For instance, in the multiclass case they're instantiating a `MulticlassAccuracy` object (https://github.com/Lightning-AI/metrics/blob/bc057d495703aff5c4d8d671c54060e08a677142/src/torchmetrics/classification/accuracy.py#L359).

**Reply by chriswilson2020** [reply]
Thanks Alessandro... I thought this course was pretty new but I guess I didn't factor in that release data does not equal creation date :D .  So much content takes time to generate of course.

**Comment by mrdbourke** [comment]
Thank you @AlessandroMiola for fixing this in #203 

I just merged the code!

Code before (errors):

```python
from torchmetrics import Accuracy
torchmetrics_accuracy = Accuracy().to(device)
```

New code: 

```python 
from torchmetrics import Accuracy
torchmetrics_accuracy = Accuracy(task='multiclass', num_classes=4).to(device)
```

See the updated `MulticlassAccuracy()` as well: https://torchmetrics.readthedocs.io/en/stable/classification/accuracy.html#multiclassaccuracy
